src/Compilers/Core/Portable/Text/LargeTextWriter.cs

```diff
@@ -154,5 +154,16 @@ private void EnsureBuffer()
                 _buffer = new char[_bufferSize];
             }
         }
+
+        protected override void Dispose(bool disposing)
```

Comment:
I moved LargeTextWriter to use Dispose as well because:
- That's the method that's supposed to do resource cleanup, and it already exists on `TextWriter` and is supposed to be called anyway
- As it is currently, it is not clear from the API that `ToSourceText` a) must be called in order to do cleanup, and b) can only be called once